### PR TITLE
Publish Singularity Image to CVMFS

### PR DESCRIPTION
**Pull request by ric-evans**

### CODE REVIEW

**Comment by ric-evans** [comment]
waiting for https://github.com/WIPACrepo/cvmfs-actions/pull/2

**Comment by ric-evans** [comment]
**TODO:** implement action for on-branch delete

**Comment by ric-evans** [comment]
@dsschult how can I make an image and commit a tag for just a branch? Where `${{ fromJSON(needs.docker.outputs.tags) }}` is a 1-element array?

**Comment by dsschult** [comment]
Note that each tag will look like 
```
icecube/skymap_scanner:<branch name or release version>
```
so you'll need to prepend `docker://`.

> @dsschult how can I make an image and commit a tag for just a branch? Where `${{ fromJSON(needs.docker.outputs.tags) }}` is a 1-element array?

For the branch case, we said we're just going to run this action manually.  It will then put the branch name in the tag.

**Comment by ric-evans** [comment]
> Note that each tag will look like
> 
> ```
> icecube/skymap_scanner:<branch name or release version>
> ```
> 
> so you'll need to prepend `docker://`.

Done

> 
> > @dsschult how can I make an image and commit a tag for just a branch? Where `${{ fromJSON(needs.docker.outputs.tags) }}` is a 1-element array?
> 
> For the branch case, we said we're just going to run this action manually. It will then put the branch name in the tag.

In other words, the array will only _be_ 1-long when triggered from a branch, yeah? With the added logic, it's good now, I think.

**Review by dsschult — COMMENTED** [review]
This generally looks good.  I'm wondering if it would be good to package this as a separate action, since I can see this being used in several places.

File: .github/workflows/publish.yml [hunk 1]
```diff
@@ -0,0 +1,89 @@
...
+          git config user.email github-actions@github.com
+
+      - name: Append Image to docker_images.txt
+        run: |
+          cd cvmfs-actions
+          for TAG in ${{ fromJSON(needs.docker.outputs.tags) }}; do
+            echo $TAG
+            # TAG  =  icecube/skymap_scanner:3
+            # LINE = docker://icecube/skymap_scanner:3 realtime/skymap_scanner:3
+            LINE="docker://$TAG $( sed "s/icecube/realtime/g" <<< $TAG )"
```
[the remark this answers is not in the document]
Comment (ric-evans) [reply .github/workflows/publish.yml]: I think it would because of the enclosing `$( )`-- but changed it anyways

**Comment by ric-evans** [comment]
> This generally looks good. I'm wondering if it would be good to package this as a separate action, since I can see this being used in several places.

An action package of just a separate yaml?

**Comment by dsschult** [comment]
> > This generally looks good. I'm wondering if it would be good to package this as a separate action, since I can see this being used in several places.
> 
> An action package or just a separate yaml?

Probably an action package.  Might as well include both jobs in it, and just specify the repo name and cvmfs dest as inputs.  cvmfs dest could even be optional, to trigger that job.

File: .github/workflows/publish.yml [hunk 2]
```diff
@@ -24,7 +28,11 @@ jobs:
         images: |
           icecube/skymap_scanner
         tags: |
-            type=ref,event=branch
+            # branches/PRs
+            type=sha,prefix={{branch}}-,enable=${{ github.ref != format('refs/heads/{0}', github.event.repository.default_branch ) }}
+            # work for master
+            type=ref,event=branch,enable={{ enable=${{ github.ref != format('refs/heads/{0}', github.event.repository.default_branch ) }} }}
```
[the remark this answers is not in the document]
Comment (ric-evans) [reply .github/workflows/publish.yml]: good catch

**Review by dsschult — APPROVED** [review]

**Pull request merged**